### PR TITLE
Support for MAX4

### PR DESCRIPTION
**Pull request by imilankovic**
Kernel changed to support MAX4 and adapted for new loop constructs like at Imperial.
Cleaned unnecessary files and added RUNRULE for Maia.

### CODE REVIEW

File: APP/RunRules/Vectis/.maxcompiler-activity.be [hunk 1]
```diff
@@ -1 +1 @@
...
...ompiler1.0ed11:CommandType4:Done7:Version14:MaxCompiler1.0ed11:CommandType4:Done7:Version14:MaxCompiler1.0ed11:CommandType4:Done7:Version14:MaxCompiler1.0ed11:CommandType4:Done7:Version14:MaxCompiler1.0ed11:CommandType4:Done7:Version14:MaxCompiler1.0ed11:CommandType16:MaxfileGenerated4:Path78:/scratch/imilankovic/maxdc_builds/19-02-16/NBody_VECTIS_DFE_/scratch/NBody.max7:Version14:MaxCompiler1.0ed11:CommandType6:Worked7:NbTasks1:17:Version14:MaxCompiler1.0ed11:CommandType4:Done7:Version14:MaxCompiler1.0ed11:CommandType16:ResultFileCopied4:Name30:NBody-NBodyKernel-original.pxg7:Version14:MaxCompiler1.0ed11:CommandType16:ResultFileCopied4:Name36:NBody-NBodyKernel-final-hardware.pxg7:Version14:MaxCompiler1.0ed11:CommandType16:ResultFileCopied4:Name7:NBody.h7:Version14:MaxCompiler1.0ed11:CommandType16:ResultFileCopied4:Name9:NBody.max7:Version14:MaxCompiler1.0ed11:CommandType12:TimingReport4:Path73:/scratch/imilankovic/maxdc_builds/19-02-16/NBody_VECTIS_DFE_/timingreport7:Version14:MaxCompiler1.0ed11:CommandType12:BuildLogDone4:Path71:/scratch/imilankovic/maxdc_builds/19-02-16/NBody_VECTIS_DFE_/_build.log7:Version14:MaxCompiler1.0ed11:CommandType10:EndOfBuild7:Version14:MaxCompiler1.0e
```
Comment (ntrifunovic) [inline APP/RunRules/Vectis/.maxcompiler-activity.be]: This file can be removed from the repo.

File: APP/RunRules/Vectis/NBody.xml [hunk 2]
```diff
@@ -29,6 +29,7 @@
             <classpath>
                 <pathelement path="${env.ANT_MAXELER_INTERNAL_MAXIDE_JARS}"/>
                 <pathelement path="../RunRules/Vectis/dist/bin"/>
+                <pathelement path="/network-raid/opt/maxq/maxq-ctl.jar"/>
```
Comment (ntrifunovic) [inline APP/RunRules/Vectis/NBody.xml]: @djovanovic-maxeler: Will this make problems for the automatic AppGallery apps testing?

There are more hardcoded paths, please also suggest changes for those if needed.

File: APP/RunRules/Vectis/.maxcompiler-activity.be (hunk 1 above)
Comment (imilankovic) [inline APP/RunRules/Vectis/.maxcompiler-activity.be]: Done.

File: APP/RunRules/Vectis/NBody.xml (hunk 2 above)
Comment (dusan-jovanovic) [inline APP/RunRules/Vectis/NBody.xml]: maxq-ctl.jar should be removed from the project.

File: APP/RunRules/Vectis/Makefile.rules [hunk 3]
```diff
@@ -62,7 +62,7 @@ include/NBody.h: maxfiles/NBody.h
 maxfiles/NBody.max maxfiles/NBody.h:
 # Ensure that maxfiles do not get build by this Makefile when running in MaxIDE
 ifndef MAXIDE_RUN
-	test -f $@ || env  MAX_BUILDPARAMS=" DFEModel=VECTIS  target=DFE  maxFileName=NBody MPPRThreads='2' MPPRStartCT='1' MPPREndCT='16' numPipes='30' "  ant -v -f NBody.xml
+	test -f $@ || env  MAX_BUILDPARAMS=" DFEModel=VECTIS  target=DFE  maxFileName=NBody MPPRThreads='2' MPPRStartCT='1' MPPREndCT='16' numPipes='30' loopLatency='12' "  ant -v -f NBody.xml
```
Comment (dusan-jovanovic) [inline APP/RunRules/Vectis/Makefile.rules]: Set the MPPRThreads to 1

File: APP/RunRules/Vectis/NBody.xml (hunk 2 above)
Comment (imilankovic) [inline APP/RunRules/Vectis/NBody.xml]: Done.

File: APP/RunRules/Vectis/Makefile.rules (hunk 3 above)
Comment (imilankovic) [inline APP/RunRules/Vectis/Makefile.rules]: Done.

**Pull request merged**